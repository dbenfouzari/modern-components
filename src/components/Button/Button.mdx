---
name: Button
---

import { Playground, PropsTable } from 'docz';
import Button from './';

# Button

<PropsTable of={Button} />

## Basic usage

<Playground>
    <Button raised={false}>Coucou</Button>
</Playground>

## Raised

<Playground>
    <Button raised={true}>Coucou</Button>
</Playground>

## With some emojis

<Playground>
    <Button>
        <span role="img" aria-label="so cool">
        😀 😎 👍 💯
        </span>
    </Button>
</Playground>
