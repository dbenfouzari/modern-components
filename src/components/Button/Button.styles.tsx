import styled, { css } from "../../styled-components";

export const StyledButton = styled.button<{ raised?: boolean }>`
  border-radius: 4px;
  border: 0 solid;
  border-bottom-width: 1px;
  box-sizing: border-box;
  color: #fff;
  font-size: 14px;
  height: 30px;
  padding: 5px 10px;
  outline: none;

  &:hover {
    border-bottom-width: 3px;
  }

  &:active {
    background-color: ${props => props.theme.primary};
  }

  ${props =>
    props.raised
      ? css`
          background-color: ${props.theme.primaryLighter};
          color: #fff;
          border-color: ${props.theme.primary};
        `
      : css`
          background-color: #fff;
          color: ${props.theme.primaryLighter};
          border: 1px solid ${props.theme.primaryLighter};

          &:active {
            color: #fff;
            background-color: ${props.theme.primaryLighter};
          }
        `}
`;
