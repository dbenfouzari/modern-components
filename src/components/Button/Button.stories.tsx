import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Button from '.';

storiesOf('Button', module)
  .add('with text', () => <Button raised={false} onClick={action('onClick')}>Coucou</Button>)
  .add('raised', () => <Button raised onClick={action('onClick')}>Coucou</Button>)
  .add(
    'with some emoji',
    () => (
      <Button>
        <span role='img' aria-label='so cool'>
          😀 😎 👍 💯
        </span>
      </Button>
    )
  )
