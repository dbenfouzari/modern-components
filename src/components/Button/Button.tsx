import * as React from "react";
import { StyledButton } from "./Button.styles";

interface ButtonProps {
  /**
   * What will be shown in the button
   *
   * @since 0.0.1
   * @version 0.0.1
   */
  children: JSX.Element | JSX.Element[] | string | number;
  /**
   * The action that will be called when user clicks
   *
   * @since 0.0.1
   * @version 0.0.1
   */
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  /**
   * Should show as raised ?
   *
   * @since 0.0.1
   * @version 0.0.1
   * @default true
   */
  raised?: boolean;
  /**
   * Which button type ?
   *
   * @since 0.0.1
   * @version 0.0.1
   * @default "button"
   */
  type?: "button" | "submit" | "reset";
}

/**
 * This is a simple button
 *
 * @since 0.0.1
 * @version 0.0.1
 * @author Donovan BENFOUZARI <d.benfouzari@gmail.com>
 */
export const Button = ({
  children,
  onClick,
  raised = true,
  type = "button"
}: ButtonProps) => (
  <StyledButton type={type} role="button" raised={raised} onClick={onClick}>
    {children}
  </StyledButton>
);

export default Button;
