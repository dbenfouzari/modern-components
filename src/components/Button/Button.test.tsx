import * as React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(<Button>Hello World!</Button>)
      .toJSON()

    expect(wrapper).toMatchSnapshot();
  });

  
})
