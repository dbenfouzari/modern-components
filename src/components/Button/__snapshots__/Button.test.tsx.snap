// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  border-radius: 2px;
  border: 0;
  color: #fff;
  font-size: 14px;
  padding: 5px 10px;
}

<button
  className="c0"
>
  Hello World!
</button>
`;
